Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs

```diff
@@ -173,6 +190,9 @@ protected virtual IWebHostBuilder CreateWebHostBuilder()
         {
             lambdaContext.Logger.LogLine($"Incoming {request.HttpMethod} requests to {request.Path}");
 
```

Comment:
You are missing an invocation to a new virtual method that you could override to set the environment variables. 

1. You need a new virtual PreStart method that takes in the ILambdaContext. I wouldn't pass in the request since this method would only be invoked the on the first request, and we want to prevent misuse of this method.

2. You need to call this method in the !IsStarted if statement. Since this method will only be useful for altering the properties of the environment before the service starts, we should isolate this call so it doesn't cause  inadvertent side  effects.